src/lib/firebase/firebaseAuth.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  createUserWithEmailAndPassword,
+  UserCredential,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+import { auth } from "./firebaseConfig";
+
+type userAuthResponse = {
+  success: boolean;
+  userId: string | null;
```

Comment:
Just letting you know, I changed this since it was small and would be quicker than asking you to change it yourself, but whenever the operation fails and success is false, it makes more sense to assign userId a value of null instead of a string because someone working with the code in the future may not know that "-1" is supposed to indicate no userId. Using null ensures that it's more likely to cause an error and is therefore easier to check for.